src/media_tools/av_parsers.c

```diff
@@ -1545,6 +1545,595 @@ GF_Err gf_m4a_write_config(GF_M4ADecSpecInfo *cfg, char **dsi, u32 *dsi_size)
+	}
+
+	if (gf_bs_read_int(bs, 1) /*operating_points_decoder_model_present*/) {
+		u8 operating_points_decoder_model_count_minus_1 = gf_bs_read_int(bs, 5);
+		size_t opNum;
+		for (opNum = 0; opNum <= operating_points_decoder_model_count_minus_1; opNum++) {
+			/*decoder_model_operating_point_idc[opNum] = */gf_bs_read_int(bs, 12);
+			Bool display_model_param_present_flag = gf_bs_read_int(bs, 1);
+			if (display_model_param_present_flag) {
+				/*initial_display_delay_minus_1 =*/ gf_bs_read_int(bs, 4);
+			}
+			if (decoder_model_info_present_flag) {
+				const Bool decoder_model_param_present_flag = gf_bs_read_int(bs, 1);
+				if (decoder_model_param_present_flag) {
+					assert(0); /*operating_parameters_info();*/
+				}
+			}
+		}
+	} else {
+		/*operating_points_decoder_model_count_minus_1 = -1;*/
+	}
+
+	/*film_grain_params_present =*/ gf_bs_read_int(bs, 1);
+}
+
+GF_Err gf_media_aom_parse_ivf_file_header(GF_BitStream *bs, AV1State *state)
+{
+	char *FourCC = NULL;
+	u32 dw = 0;
+	
+	if (gf_bs_available(bs) < 32) {
+		GF_LOG(GF_LOG_ERROR, GF_LOG_CODING, ("[IVF] Not enough bytes available ("LLU").\n", gf_bs_available(bs)));
+		return GF_NON_COMPLIANT_BITSTREAM;
+	}
+
+	dw = gf_bs_read_u32(bs);
+	if (dw != GF_4CC('D', 'K', 'I', 'F')) {
+		GF_LOG(GF_LOG_ERROR, GF_LOG_CODING, ("[IVF] Invalid signature\n"));
+		return GF_NON_COMPLIANT_BITSTREAM;
+	}
+
+	dw = gf_bs_read_u16_le(bs);
+	if (dw != 0) {
+		GF_LOG(GF_LOG_ERROR, GF_LOG_CODING, ("[IVF] Wrong IVF version. 0 expected, got %u\n", dw));
+		return GF_NON_COMPLIANT_BITSTREAM;
+	}
+	dw = gf_bs_read_u16_le(bs);
+
+	dw = gf_bs_read_u32(bs); //codec_fourcc
+	FourCC = (char*)&dw;
+	if (dw != GF_4CC('A', 'V', '0', '1')) {
+		GF_LOG(GF_LOG_ERROR, GF_LOG_CODING, ("[IVF] Wrong codec FourCC. Only 'AV01' supported, got '%c%c%c%c'\n", FourCC[3], FourCC[2], FourCC[1], FourCC[0]));
+		return GF_NON_COMPLIANT_BITSTREAM;
+	}
+	dw = gf_bs_read_u16_le(bs);
+	state->width = state->width < dw ? dw : state->width;
+	dw = gf_bs_read_u16_le(bs);
+	state->height = state->height < dw ? dw : state->height;
+	dw = gf_bs_read_u32_le(bs); //time_base.numerator
+	dw = gf_bs_read_u32_le(bs); //time_base.denominator
+
+	gf_bs_read_u64(bs); //skip
+
+	return GF_OK;
+}
+
+GF_Err gf_media_aom_parse_ivf_frame_header(GF_BitStream *bs, u64 *frame_size)
+{
+	assert(frame_size);
+	*frame_size = gf_bs_read_u32_le(bs);
+	if (*frame_size > 256 * 1024 * 1024) {
+		GF_LOG(GF_LOG_ERROR, GF_LOG_CODING, ("[IVF] Wrong frame size %u\n", *frame_size));
+		*frame_size = 0;
+		return GF_NON_COMPLIANT_BITSTREAM;
+	}
+
+	gf_bs_read_u64(bs); //skip
+
+	return GF_OK;
+}
+
+static GF_Err av1_parse_obu_header(GF_BitStream *bs, ObuType *obu_type, Bool *obu_extension_flag, Bool *obu_has_size_field, u8 *temporal_id, u8 *spatial_id)
+{
+	Bool forbidden = gf_bs_read_int(bs, 1);
+	if (forbidden) {
+		return GF_NON_COMPLIANT_BITSTREAM;
+	}
+	assert(obu_type && obu_extension_flag && obu_has_size_field && temporal_id && spatial_id);
```

Comment:
Why? This will certainly fail for typical bitstream (without extension flag for example).